specs/core/0_beacon-chain.md

```diff
@@ -1323,34 +1295,32 @@ def process_justification_and_finalization(state: BeaconState) -> None:
 
     # Process justifications
     state.previous_justified_checkpoint = state.current_justified_checkpoint
-    state.justification_bitfield = (state.justification_bitfield << 1) % 2**64
-    previous_epoch_matching_target_balance = get_attesting_balance(
-        state, get_matching_target_attestations(state, previous_epoch)
-    )
-    if previous_epoch_matching_target_balance * 3 >= get_total_active_balance(state) * 2:
+    state.justification_bits[1:] = state.justification_bits[:JUSTIFICATION_BITS_LENGTH - 1]
```

Comment:
If we are comfortable using python array slices here, I'd say we go all the way and use the `-1` notation.
```suggestion
    state.justification_bits[1:] = state.justification_bits[:-1]
```

Reply:
could also do

```suggestion
    state.justification_bits = [0b0] + state.justification_bits[:-1]
```

Reply:
Did that and @dankrad mentioned this makes `justification_bits` a list (as opposed to a bitvector). The tests didn't complain though.

Reply:
I think the assignment can handle the cast

Reply:
We could probably overload the `1` operator to do it but currently I think it doesn't:

```python
>>> a = ssz_typing.Bitvector[2](0, 0)
>>> b = [0] + a
>>> type(b)
<class 'list'>
```